src/classic/clvm_tools/stages/stage_2/operators.rs

```diff
@@ -31,6 +31,7 @@ pub struct CompilerOperators {
     compile_outcomes: RefCell<HashMap<String, String>>,
     dialect: RefCell<Rc<dyn Dialect>>,
     runner: RefCell<Rc<dyn TRunProgram>>,
+    opt_memo: RefCell<HashMap<String, NodePtr>>,
```

Comment:
IMO it's not at all weird to use atomics as module level counters for things like diagnostic indent level, but I suppose it's taste.  I removed it.
The string is a treehash of the code.  Makes sense to have a wrapper type for that I suppose.

Reply:
hmm remembered that I had used allocator keys in this map as well as a way to avoid doing sha256tree if it's avoidable. The memo is shared among a bunch of independent interpreter invocations in the same stack, but that stack will never use a different allocator (otherwise everything would be broken).  Good idea to separate that into a proper enum key.

Reply:
and also realized that reordering this allows me to save the treehash generation here too some of the time.